VDYP-167 Processing FIP Polygons/Layers/Species into VDYP Polygons/Layers/Species

smithkm opened the pull request.

mjunkin commented on vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/LazyValue.java:
```diff
@@ -0,0 +1,37 @@
+		return value;
+	}
+
+	public LazyValue(Supplier<T> compute) {
```
Constructors are normally at the start of the list of methods

mjunkin commented on vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/Utils.java:
```diff
@@ -23,4 +31,49 @@ public static <T> Set<T> singletonOrEmpty(@Nullable T value) {
 		return Collections.singleton(value);
 	}
 
+	@SuppressWarnings("unchecked")
```
Please add javadoc for this method

smithkm replied:
Done

mjunkin commented on vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/ValueOrMarker.java:
```diff
@@ -0,0 +1,123 @@
+	}
+
+	/**
+	 * Is there a value
```
@return true if and only if this ValueOrMarker is a marker.

smithkm replied:
Yes, I got those backward.

mjunkin commented on vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/ValueOrMarker.java:
```diff
@@ -0,0 +1,123 @@
+	}
+
+	/**
+	 * Is there a marker
```
@return true if and only if this ValueOrMarker is a value.

mjunkin commented on vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/ValueOrMarker.java:
```diff
@@ -0,0 +1,123 @@
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	public Optional<Value> getValue() {
```
Shouldn't getValue throw when the MarkerOrValue is not a value, since Optional.empty() is a possible value()?

smithkm replied:
No, that's why it returns Optional.

mjunkin commented on vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/ValueOrMarker.java:
```diff
@@ -0,0 +1,123 @@
+	 */
+	@SuppressWarnings("unchecked")
+	public Optional<Marker> getMarker() {
+		if (isMarker()) {
```
Shouldn't this throw when it's not a Marker?

smithkm replied:
The value and the marker are `Optional` to represent this. The calling code can `.orElseThrow(...)` to throw an exception or choose to handle it some other way.   A `ValueOrMarker` will always have one of them present and the other empty.

mjunkin commented:
This review is not complete, but it appears the github requires that you commit your review comments.

smithkm commented:
@mjunkin OK I think I've addressed all your concerns that are specific to this PR.  VDYP-163/164/165 issues I'll tacking with a new PR once everything is merged as those PRs are merged.  VDYP-168 and 206 issues I'll take care of in their respective PRs.

smithkm commented on vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/HoldFirst.java:
```diff
@@ -0,0 +1,44 @@
+	 * Sets the value by calling the supplier if it has not been set already.
+	 * Returns the value.
+	 */
+	public T get(Supplier<T> source) {
```
No it should be ignored silently.  Docs updated.
